enterprise/frontend/src/metabase-enterprise/dependencies/components/DependencyList/ListSidebar/utils.ts

```diff
@@ -0,0 +1,21 @@
+import type { DependencyError, DependencyErrorType } from "metabase-types/api";
+
+import type { DependencyErrorGroup } from "./types";
+
+export function getDependencyErrorGroups(
+  errors: DependencyError[],
```

Comment:
I'm wondering if you should change this to 

```suggestion
  errors: [DependencyError, ...DependencyError[]],
```

(or some other way to specify that it has to have at least one error / cannot be empty)

because you do `errors[0].type` below which means that it has to be non-empty, right?